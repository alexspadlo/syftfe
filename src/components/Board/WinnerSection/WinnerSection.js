import React from 'react'
import { resetBoard } from '../../../store/actions/moves'
import { useDispatch } from 'react-redux'
import PropTypes from 'prop-types'
import winnerImage from '../../../static/winner.png'
import opsImage from '../../../static/ops.png'
import s from './WinnerSection.module.scss'

const WinnerSection = ({winner}) => { 
    const dispatch = useDispatch()
    if(!winner) return null   
    return (
        <section className={`w-100 ${s.containerBack}`}>
            <div className="container">
                {winner === 'Tie'
                    ? <div className="row m-0">
                        <div className={`col-12 ${s.customPaddingTop} text-center`}>
                            <h1 className={s.customTitle}>OOPPS... IT'S A TIE !!!</h1>
                            <h2 className={`${s.customSubtitle} pt-5`}> TRY AGAIN</h2>
                        </div>
                        <div className={`col-lg-4 col-12 ${s.customPaddingBottom} m-auto`}>
                            <img className="img-fluid" src={opsImage} alt="tictoc" />
                        </div>
                        <section className="w-100 pb-5 text-center">
                            <button onClick={() => dispatch(resetBoard())} className={s.customButton}>TRY AGAIN</button>
                        </section>
                    </div>
                    :
                    <div className="row m-0">
                        <div className={`col-12 ${s.customPaddingTop} text-center`}>
                            <h1 className={s.customTitle}>AND THE WINNER IS....</h1>
                            <h2 className={`${s.customSubtitle} pt-5`}>PLAYER: {winner}</h2>
                        </div>
                        <div className={`col-lg-4 col-12 ${s.customPaddingBottom} m-auto`}>
                            <img className="img-fluid" src={winnerImage} alt="tictoc" />
                        </div>
                        <section className="w-100 pb-5 text-center">
                            <button onClick={() => dispatch(resetBoard())} className={s.customButton}>TRY AGAIN</button>
                        </section>
                    </div>
                }
            </div>
        </section>
    )
}

WinnerSection.propTypes = {
    winner: PropTypes.string
}

WinnerSection.defaultProps = {
	winner: null
}

export default WinnerSection