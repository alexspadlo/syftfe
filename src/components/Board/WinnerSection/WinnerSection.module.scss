.containerBack {
  background-size: cover;
  background-color: rgb(160, 81, 136);
  min-height: 100vh;
  height: 100%;

  .customPaddingTop {
    padding-top: 70px;
    padding-bottom: 30px;

    .customTitle {
      font-size: 50px;
      color: white;
    }

    .customSubtitle {
      font-size: 30px;
      color: white;
    }
  }

  .customPaddingBottom {
    padding-bottom: 70px;
  }

  .customButton {
    background-color: rgb(226, 75, 100);
    border-radius: 20px;
    border: 1px solid rgb(214, 78, 100);
    padding: 10px 30px;
    color: white;

    &:focus {
      outline: 0;
    }
  }
}
