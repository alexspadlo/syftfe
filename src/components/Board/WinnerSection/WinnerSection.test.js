import React from 'react'
import { render } from '@testing-library/react'
import WinnerSection from './WinnerSection'
import configureStore from '../../../store'
import { Provider } from 'react-redux'

const store = configureStore()

describe('Testing WinnerSection Component', () => {
	test('Do not render WinnerSection without the props', () => {
		const { container } = render(
			<Provider store={store}>
				<WinnerSection />
			</Provider>
		)
		expect(container.firstChild).toBeNull()
    })
    test('Should render the winner X', () => {
		const { container, getByText } = render(
			<Provider store={store}>
				<WinnerSection winner={'X'} />
			</Provider>
        )
        expect(getByText('PLAYER: X')).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot()  
    })
    test('Should render the winner O', () => {
		const { container, getByText } = render(
			<Provider store={store}>
				<WinnerSection winner={'O'} />
			</Provider>
        )
        expect(getByText('PLAYER: O')).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot()  
    })
    test('Should render the Tie section', () => {
		const { container, getByText } = render(
			<Provider store={store}>
				<WinnerSection winner={'Tie'} />
			</Provider>
        )
        expect(getByText(`OOPPS... IT'S A TIE !!!`)).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot()  
	})
})