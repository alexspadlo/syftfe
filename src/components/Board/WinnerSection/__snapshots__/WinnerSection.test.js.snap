// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing WinnerSection Component Should render the Tie section 1`] = `
<section
  class="w-100 containerBack"
>
  <div
    class="container"
  >
    <div
      class="row m-0"
    >
      <div
        class="col-12 customPaddingTop text-center"
      >
        <h1
          class="customTitle"
        >
          OOPPS... IT'S A TIE !!!
        </h1>
        <h2
          class="customSubtitle pt-5"
        >
           TRY AGAIN
        </h2>
      </div>
      <div
        class="col-lg-4 col-12 customPaddingBottom m-auto"
      >
        <img
          alt="tictoc"
          class="img-fluid"
          src="ops.png"
        />
      </div>
      <section
        class="w-100 pb-5 text-center"
      >
        <button
          class="customButton"
        >
          TRY AGAIN
        </button>
      </section>
    </div>
  </div>
</section>
`;

exports[`Testing WinnerSection Component Should render the winner O 1`] = `
<section
  class="w-100 containerBack"
>
  <div
    class="container"
  >
    <div
      class="row m-0"
    >
      <div
        class="col-12 customPaddingTop text-center"
      >
        <h1
          class="customTitle"
        >
          AND THE WINNER IS....
        </h1>
        <h2
          class="customSubtitle pt-5"
        >
          PLAYER: 
          O
        </h2>
      </div>
      <div
        class="col-lg-4 col-12 customPaddingBottom m-auto"
      >
        <img
          alt="tictoc"
          class="img-fluid"
          src="winner.png"
        />
      </div>
      <section
        class="w-100 pb-5 text-center"
      >
        <button
          class="customButton"
        >
          TRY AGAIN
        </button>
      </section>
    </div>
  </div>
</section>
`;

exports[`Testing WinnerSection Component Should render the winner X 1`] = `
<section
  class="w-100 containerBack"
>
  <div
    class="container"
  >
    <div
      class="row m-0"
    >
      <div
        class="col-12 customPaddingTop text-center"
      >
        <h1
          class="customTitle"
        >
          AND THE WINNER IS....
        </h1>
        <h2
          class="customSubtitle pt-5"
        >
          PLAYER: 
          X
        </h2>
      </div>
      <div
        class="col-lg-4 col-12 customPaddingBottom m-auto"
      >
        <img
          alt="tictoc"
          class="img-fluid"
          src="winner.png"
        />
      </div>
      <section
        class="w-100 pb-5 text-center"
      >
        <button
          class="customButton"
        >
          TRY AGAIN
        </button>
      </section>
    </div>
  </div>
</section>
`;
